src/spaces/utilities.jl

```diff
@@ -55,6 +55,29 @@ function edistance(p1::ValidPos, p2::ValidPos, model::ABM{<:GridSpace{D,true}})
     sqrt(total)
 end
 
+"""
+    get_direction(from, to, model::ABM)
+Return the direction vector from `from` to `to` taking into account periodicity of the space
```

Comment:
Return the direction vector from the position `from` to position `to`

since we don't explicitly say what from and to are.

OR you could add an agent wrapper here as well such that `get_direction(from.pos, to.pos, model.space)` is satisfied.